### PR TITLE
Add the Oil Derrick count to the economy statistics

### PR DESCRIPTION
**Pull request by abcdefg30**
Only RA and TD have Oil Derricks.

### CODE REVIEW

**Comment by pchote** [comment]
I'm not sure that blindly counting `CashTrickler`s is a good idea. Would be better IMO if we added a tag trait for this.

**Comment by GraionDilach** [comment]
Agreeing with the tag trait - an "Oil Refinery" usecase to increase the amount of cash produced from the Oil Derricks via flipping between active CashTricklers would already give this method an issue.

**Comment by abcdefg30** [comment]
Updated.

**Comment by pchote** [comment]
I guess we should include this in the next playtest, so adding to the milestone.

**Review by teinarss — DISMISSED** [review]

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/ObserverStatsLogic.cs [hunk 1]
```diff
@@ -360,6 +360,10 @@ ScrollItemWidget EconomyStats(Player player)
 			var harvesters = template.Get<LabelWidget>("HARVESTERS");
 			harvesters.GetText = () => world.ActorsHavingTrait<Harvester>().Count(a => a.Owner == player && !a.IsDead).ToString();
 
+			var cashTricklers = template.GetOrNull<LabelWidget>("DERRICKS");
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/ObserverStatsLogic.cs]: `var derricks`?

File: mods/ra/chrome/ingame-observer.yaml [hunk 2]
```diff
@@ -412,6 +412,14 @@ Container@OBSERVER_WIDGETS:
 									Text: Harvesters
 									Align: Right
 									Shadow: True
+								Label@DERRICKS_HEADER:
```
Comment (pchote) [inline mods/ra/chrome/ingame-observer.yaml]: This makes the panel too wide at our minimum horizontal resolution (1024px):

Screenshot 2019-09-06 at 17 10 02

The "Earned this min" "Harvesters" and "Oil Derricks" columns are clearly far too wide for their data, but I don't have a simple solution to reduce their headers...

**Comment by pchote** [comment]
If we can't do a quick fix (like removing "Earned this min") then we may need to drop this from the milestone until someone can relabel / redesign the headers to not be constrained by these long labels.

File: mods/ra/chrome/ingame-observer.yaml (hunk 2 above)
Comment (tovl) [reply mods/ra/chrome/ingame-observer.yaml]: "Harvesters" and "Oil Derricks" could be replaced by pictograms. That's not a short term solution though, unless someone already has such pictograms lying around.

Comment (GraionDilach) [reply mods/ra/chrome/ingame-observer.yaml]: Think they already have customized GPS dots, which can pass as such.

**Comment by abcdefg30** [comment]
We could gain some space by using something like `$ this min` (we have `$/min` as well), but I don't think it would be enough. The question is if earned this minute is even that useful, since you kind of already have the graph displaying that information(?). (Would work good with #17044 imho.)

**Comment by abcdefg30** [comment]
Updated.

**Comment by pchote** [comment]
LGTM, although I did notice #17059 while testing this. Can you please merge the last two commits?

**Comment by abcdefg30** [comment]
Rebased and squashed.

**Comment by yaaaaa** [comment]
Hmm... why statistic needs oil derrick count? oil derrick count, and its plasements is static. The maps have own oil derrick counts, and name of The Map is exist and most importent parametr.

**Review by pchote — APPROVED** [review]

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/826eeb890cd0c340c3b9bafff0080aa1b21fc7d8...519325b9379b5ac9e4a848e3af9243efcceffb64

https://github.com/OpenRA/OpenRA/commit/380da37132723af68f95b9f3b3ff56d357eb7504

**Pull request merged**